File: mergeSort.js
function mergeShort(array){
    if (array.length>1){
        var mid=array.length/2;
        var LeftPart=array.slice(0,mid);
        var RightPart=array.slice(mid,array.length);

        mergeShort(LeftPart)
        mergeShort(RightPart)
        var i=0;
        var j=0;
        var k=0;
        while (i<LeftPart.length && j<RightPart.length){
            if (LeftPart[i]<RightPart[j]){
                array[k]=LeftPart[i];
                i++;
            }else{
                array[k]=RightPart[j];
                j++;
            } k++;
        }
        while (i<LeftPart.length){
            array[k]=LeftPart[i];
            i++;
            k++;
        }
        while (j<RightPart.length){
            array[k]=RightPart[j];
            j++;
            k++;
        }   
    return array;
    }
    
}
console.log(mergeShort([1,3,2,4]))

File: mergeSort.py
def mergeSort(array): 
    if len(array) >1: 
        mid = len(array)/2
        L = array[:mid] 
        R = array[mid:]
        mergeSort(L) 
        mergeSort(R) 
        i = j = k = 0
        while i < len(L) and j < len(R): 
            if L[i] < R[j]: 
                array[k] = L[i] 
                i+=1
            else: 
                array[k] = R[j] 
                j+=1
            k+=1
        while i < len(L): 
            array[k] = L[i] 
            i+=1
            k+=1
        while j < len(R): 
            array[k] = R[j] 
            j+=1
            k+=1
    return array
arr = [12, 11, 13, 5, 6, 7]  
print(mergeSort(arr))